### PR TITLE
Add MotionLayout light to Compose

### PR DESCRIPTION
**Pull request by camaelon**
add MotionLayout "light" composable (no keyframes, linear interpolation between two constraintsets)

add support for rotation/translation/scale/alpha in state & json
add WidgetFrame in core
add examples ScreenExample7 & 8

### CODE REVIEW

**Review by jafu888 — APPROVED** [review]

File: constraintlayout/compose/src/main/java/androidx/constraintlayout/compose/ConstraintLayout.kt [hunk 1]
```diff
@@ -186,6 +236,35 @@ internal fun rememberConstraintLayoutMeasurePolicy(
...
 
+@Composable
+@PublishedApi
+internal fun rememberMotionLayoutMeasurePolicy(
+    optimizationLevel: Int,
+    constraintSetStart: ConstraintSet,
+    constraintSetEnd: ConstraintSet,
+    progress: Float,
+    measurer: MotionMeasurer
+) = remember(optimizationLevel, constraintSetStart, constraintSetEnd, progress) {
```
Comment (mihaipopa12) [inline constraintlayout/compose/src/main/java/androidx/constraintlayout/compose/ConstraintLayout.kt:247]: I think it's better to remove progress from the remember here, as this way the MeasurePolicy will be recreated on every frame.
I would do it like this on lines 196:
```
val progressState = remember { mutableStateOf(0f) }
SideEffect { progressState.value = progress }
```
Then pass this state to `rememberMotionLayoutMeasurePolicy` and use `progressState.value` as parameters to `performInterpolationMeasure` and `performLayout`

Comment (mihaipopa12) [reply constraintlayout/compose/src/main/java/androidx/constraintlayout/compose/ConstraintLayout.kt:247]: You might also need to move measurer.clear somewhere else.

File: constraintlayout/compose/src/main/java/androidx/constraintlayout/compose/ConstraintLayout.kt [hunk 2]
```diff
@@ -1454,17 +1535,21 @@ internal class Measurer : BasicMeasure.Measurer, DesignInfoProvider {
             true
         }
         MATCH_CONSTRAINT -> {
-            Log.d("CCL2", "Measure strategy ${measureStrategy}")
-            Log.d("CCL2", "DW ${matchConstraintDefaultDimension}")
-            Log.d("CCL2", "ODR ${otherDimensionResolved}")
-            Log.d("CCL2", "IRH ${currentDimensionResolved}")
+            if (DEBUG) {
+                Log.d("CCL2", "Measure strategy ${measureStrategy}")
```
Comment (mihaipopa12) [inline constraintlayout/compose/src/main/java/androidx/constraintlayout/compose/ConstraintLayout.kt:1539]: nit: I think these can be removed.

File: constraintlayout/compose/src/main/java/androidx/constraintlayout/compose/ConstraintLayout.kt [hunk 3]
```diff
@@ -164,6 +171,49 @@ inline fun ConstraintLayout(
     )
 }
 
+enum class MotionLayoutDebugFlags {
```
[the remark this answers is not in the document]
Comment (camaelon) [reply constraintlayout/compose/src/main/java/androidx/constraintlayout/compose/ConstraintLayout.kt:174]: yep, will do in another CL

File: constraintlayout/compose/src/main/java/androidx/constraintlayout/compose/ConstraintLayout.kt (hunk 1 above)
Comment (camaelon) [reply constraintlayout/compose/src/main/java/androidx/constraintlayout/compose/ConstraintLayout.kt:250]: good idea, done.

**Pull request merged**